libsolidity/formal/VariableUsage.cpp

```diff
@@ -30,8 +30,25 @@ void VariableUsage::endVisit(Identifier const& _identifier)
 	Declaration const* declaration = _identifier.annotation().referencedDeclaration;
 	solAssert(declaration, "");
 	if (VariableDeclaration const* varDecl = dynamic_cast<VariableDeclaration const*>(declaration))
-		if (_identifier.annotation().lValueRequested)
```

Comment:
Could keep that check (i.e. do it first) because the other checks might be expensive.